### PR TITLE
Make sure sanitizeUrl method allows array in $_GET, and make methods protected to be reused

### PR DESCRIPTION
**Pull request by PierreRambaud**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | If the url contains arrays it raises an error. Make sure the value is properly checked. Also change the scope of methods to be reused due to inheritance.
| Type?         | bug fix
| Category?     | FO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | See below.

Before: 
With this URL: http://ps-develop.localhost/gb/index.php?controller=product&token=e5db2945960c5fbc72dda674efa718fd&id_product=1&id_customization=0&group%5B1%5D=1&group%5B2%5D=10&qty=1
image

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/20773)

### CODE REVIEW

File: classes/controller/FrontController.php [hunk 1]
```diff
@@ -1991,18 +1991,38 @@ private function sanitizeUrl(string $url): string
...
+            $params[Tools::safeOutput($key)] = is_array($value) ? array_walk_recursive($value, 'Tools::safeOutput') : Tools::safeOutput($value);
         }
 
         $str_params = http_build_query($params, '', '&');
-        if (!empty($str_params)) {
-            $sanitizedUrl = preg_replace('/^([^?]*)?.*$/', '$1', $url) . '?' . $str_params;
-        } else {
-            $sanitizedUrl = preg_replace('/^([^?]*)?.*$/', '$1', $url);
-        }
+        $sanitizedUrl = preg_replace('/^([^?]*)?.*$/', '$1', $url) . (!empty($str_params) ? '?' . $str_params : '');
```
Comment (PierreRambaud) [inline classes/controller/FrontController.php:2005]: Do not repeat the regex!

**Review by NeOMakinG — APPROVED** [review]
Oh yeah

**Review by matthieu-rolland — APPROVED** [review]

**Comment by ngodefroy** [comment]
TEST OK 
I submitted the link attached: /index.php?controller=product&token=e5db2945960c5fbc72dda674efa718fd&id_product=1&id_customization=0&group%5B1%5D=1&group%5B2%5D=10&qty=1

returned 
Capture d’écran 2020-09-02 à 12 35 30
OK

**Comment by Progi1984** [comment]
Thanks @PierreRambaud & @ngodefroy

**Pull request merged**